assembl/static/js/app/models/attachments.js

```diff
@@ -370,8 +393,152 @@ var AttachmentCollection = Base.Collection.extend({
   }
 });
 
+/*
+  An attachment collection that allows for validation
+ */
+var ValidationAttachmentCollection = AttachmentCollection.extend({
+
+
+  initialize: function(models, options){
+    this.limits = options.limits || {};
+    AttachmentCollection.prototype.initialize.apply(this, arguments);
+  },
+
+  /*
+    takes an array of models and makes validation check
+   */
+  addValidation: function(models){
+    //If there is a count limit, override the old data (remove them first, though)
+    //The removal is done in a FIFO format
+
+    if (!models){
+      return [];
+    }
+
+    var correctedNumberModelsCollection = models,
+        that = this;
+
+    if (!this.isCountLimitCorrect(models)){
+      correctedNumberModelsCollection= this.getCorrectCountedCollection(models);
+      if (this.length > models.length){
+        for (var i in correctedNumberModelsCollection.count){
+          var modelToRemove = this.shift();
+          modelToRemove.destroy();
+        }
+      }
+      else {
+        this.each(function(model){
+          model.destroy();
+        });
+      }
+
+      correctedNumberModelsCollection = correctedNumberModelsCollection.collection;
+    }
+
+    _.each(correctedNumberModelsCollection, function(model){
+      if (!that.isTypeLimitCorrect(model)){
+        model.typeLimitReached = true;
+        Growl.showBottomGrowl(
```

Comment:
I would have liked that. But, at the time of the attachment model creation, the attachment view for that model is not yet created. So, during this check, the view isn't created. Unless we have a parent Event Aggregator at the topmost view, and THAT view would listen for an event from these models, and that view throws a growl.
This was a quick win to show a growl from the model, given that the growl is not bound to the attachment view. Please discuss.